### ISSUE TITLE
ALE not receiving any output from ESLint

### ISSUE THREAD

**Issue by wizardlink** [dense-analysis/ale#2878]
## Information

```
VIM - Vi IMproved 8.1 (2018 May 18, compiled Oct 30 2019 23:02:18)
MS-Windows 64-bit GUI version with OLE support
Included patches: 1-2234
```

Operating System: 

## What went wrong

Basically, ALE is not getting any information from ESLint. I looked into the `:ALEInfo` to find what command it was running, and tried to do it myself on the directory that I had the file on. Which that worked perfectly fine, but for some reason no output was received by ALE. All that I can assume is that it try to run out of scope (while being in a different directory, thus ESLint not being able to find my plugins in node_modules).

### :ALEInfo

```

 Current Filetype: typescript
Available Linters: ['eslint', 'tslint', 'tsserver', 'typecheck', 'xo']
  Enabled Linters: ['eslint', 'tsserver']
 Suggested Fixers: 
  'eslint' - Apply eslint --fix to a file.
  'prettier' - Apply prettier to a file.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
  'tslint' - Fix typescript files with tslint --fix.
  'xo' - Fix JavaScript/TypeScript files using xo --fix.
 Linter Variables:

let g:ale_typescript_tsserver_config_path = ''
let g:ale_typescript_tsserver_executable = 'tsserver'
let g:ale_typescript_tsserver_use_global = 0
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = 100
let g:ale_completion_enabled = 1
let g:ale_completion_max_suggestions = 50
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 1
let g:ale_fixers = {'typescript': ['eslint'], 'json': ['prettier'], '*': ['remove_trailing_lines', 'trim_whitespace'], 'javascript': ['eslint']}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 300
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {'typescript': ['eslint', 'tsserver'], 'javascript': ['eslint']}
let g:ale_linters_explicit = 1
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_lsp_root = {}
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 1
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>>'
let g:ale_sign_info = '--'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = '>>'
let g:ale_sign_style_warning = '--'
let g:ale_sign_warning = '--'
let g:ale_sign_highlight_linenrs = 0
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - success) E:\Hydractify\Projects\Kanna Kobayashi\GitHub\node_modules/eslint/bin/eslint.js
(finished - exit code 2) 'cmd /s/c "node.exe "E:\Hydractify\Projects\Kanna Kobayashi\GitHub\node_modules/eslint/bin/eslint.js" -f json --stdin --stdin-filename "E:\Hydractify\Projects\Kanna Kobayashi\GitHub\src\util\paginate.ts" < C:\Users\Usuario\AppData\Local\Temp\VIB2113.tmp\paginate.ts"'

<<<NO OUTPUT RETURNED>>>

(executable check - success) E:\Hydractify\Projects\Kanna Kobayashi\GitHub\node_modules/.bin/tsserver
(started) 'cmd /s/c ""E:\Hydractify\Projects\Kanna Kobayashi\GitHub\node_modules/.bin/tsserver""'
(finished - exit code 2) 'cmd /s/c "node.exe "E:\Hydractify\Projects\Kanna Kobayashi\GitHub\node_modules/eslint/bin/eslint.js" -f json --stdin --stdin-filename "E:\Hydractify\Projects\Kanna Kobayashi\GitHub\src\util\paginate.ts" < C:\Users\Usuario\AppData\Local\Temp\VIC16C5.tmp\paginate.ts"'

<<<NO OUTPUT RETURNED>>>

(finished - exit code 2) 'cmd /s/c "node.exe "E:\Hydractify\Projects\Kanna Kobayashi\GitHub\node_modules/eslint/bin/eslint.js" -f json --stdin --stdin-filename "E:\Hydractify\Projects\Kanna Kobayashi\GitHub\src\util\paginate.ts" < C:\Users\Usuario\AppData\Local\Temp\VID1975.tmp\paginate.ts"'

<<<NO OUTPUT RETURNED>>>
```

**Comment by wizardlink** [comment]
After fixing all the `eslint` issues myself, it seems to lint properly now for some odd reason.

I will close the issue, but might re-open in case it occurs again.

**Comment by wizardlink** [comment]
So it seems the issue is still happening, I am not quite sure what made it work the last time, this really does not makes sense. I have made the same tests, `eslint` is working, but ALE might be trying to run out of scope?

> *Observation: I did try to do `:ALELint`, it still didn't properly lint.*

**Comment by hsanson** [comment]
@wizardlink can you test if this issue still happens with latest master? Recently a fix for eslint #2891 was merged.

**Comment by wizardlink** [comment]
> 
> 
> @wizardlink can you test if this issue still happens with latest master? Recently a fix for eslint #2891 was merged.

It seems to have been fixed. I can not go through my workflow to fully test it, but if anything I re-open the issue and take a deeper look into it. Looking at the PR though, I doubt that my problem was not solved, but who knows. Thank you!
